src/Utilities/model.jl

```diff
@@ -691,15 +691,37 @@ _callfield(f, s::SymbolFS) = :($f(model.$(_field(s))))
 _broadcastfield(b, s::SymbolFS) = :($b(f, model.$(_field(s))))
 
 """
-macro model(modelname, scalarsets, typedscalarsets, vectorsets, typedvectorsets, scalarfunctions, typedscalarfunctions, vectorfunctions, typedvectorfunctions)
+    macro model(modelname, scalarsets, typedscalarsets, vectorsets, typedvectorsets, scalarfunctions, typedscalarfunctions, vectorfunctions, typedvectorfunctions)
```

Comment:
Can we provide a default instantiation of this type that supports *all* functions and sets that are defined in MOI? I think this would cover a lot of the existing use cases, and would save people from having to call the macro.

Reply:
We already provide `JuMP._MOIModel` which I use most of the time

Reply:
That's not a public symbol, and it requires depending on JuMP.

Reply:
We could define a `@model(Model, ...)` which is the same as JuMP's `@model`. Then `JuMP` won't have to define `_MOIModel`, it can just use `MOI.Model`, which decrease the chance of breaking JuMP with MOI.
I would be ok with that.

Reply:
> We could define a `@model(Model, ...)` which is the same as JuMP's `@model`. Then JuMP won't have to define _MOIModel, it can just use MOI.Model, which decrease the chance of breaking JuMP with MOI.
I would be ok with that.

I support this. I would call it something like `DefaultModel` or `BasicModel` to make it clear that it's possible to have different implementations of ` ModeLike`. The `@model` macro then becomes a footnote, because most use cases are covered by `BasicModel`.